sdks/java/harness/src/main/java/org/apache/beam/fn/harness/data/PCollectionConsumerRegistry.java

```diff
@@ -209,44 +235,59 @@ public Map<String, ByteString> getExecutionTimeMonitoringData(ShortIdMap shortId
   private class MetricTrackingFnDataReceiver<T> implements FnDataReceiver<WindowedValue<T>> {
     private final FnDataReceiver<WindowedValue<T>> delegate;
     private final SimpleExecutionState state;
-    private final Counter unboundedElementCountCounter;
-    private final SampleByteSizeDistribution<T> unboundedSampledByteSizeDistribution;
+    private final BundleCounter elementCountCounter;
+    private final SampleByteSizeDistribution<T> sampledByteSizeDistribution;
     private final Coder<T> coder;
     private final MetricsContainer metricsContainer;
 
     public MetricTrackingFnDataReceiver(
-        String pCollectionId, ConsumerAndMetadata consumerAndMetadata) {
+        String pCollectionId, Coder<T> coder, ConsumerAndMetadata consumerAndMetadata) {
       this.delegate = consumerAndMetadata.getConsumer();
       this.state = consumerAndMetadata.getExecutionState();
-      HashMap<String, String> labels = new HashMap<String, String>();
-      labels.put(Labels.PCOLLECTION, pCollectionId);
-
-      // Collect the metric in a metric container which is not bound to the step name.
-      // This is required to count elements from impulse steps, which will produce elements outside
-      // of a pTransform context.
-      MetricsContainer unboundMetricContainer = metricsContainerRegistry.getUnboundContainer();
 
+      HashMap<String, String> labels = new HashMap<>();
+      labels.put(Labels.PCOLLECTION, pCollectionId);
```

Comment:
nit: looks like this line didn't change at all but the surrounding whitespace did...

Reply:
I grouped together the logical portion of creating this counter which was broken up before.